pkg/plugin/plugin.go

```diff
@@ -164,13 +166,27 @@ func (p *Plugin) Health() error {
 		return err
 	}
 	if err != nil {
-		zap.L().Warn("health check fail", zap.Error(err))
+		zap.L().Warn("health check failed", zap.Error(err))
 	} else {
 		zap.L().Debug("health check success")
 	}
 	return err
 }
 
+// Checks the liveness of KMS API.
+// If the error returned from KMS is user-induced, the function returns nil.
+func (p *Plugin) Check() error {
+	err := p.Health()
+	if err == nil {
+		return nil
+	}
+	et := ParseError(err)
+	if et == KMS_ERROR_TYPE_USER_INDUCED {
+		return nil
+	}
+	return err
```

Comment:
This reads a little unusual, checking for a nil error first.

```suggestion
	if err := p.Health(); err != nil && ParseError(err) != KMS_ERROR_TYPE_USER_INDUCED {
    	    return err
	}
	return nil
```